download-utils/src/lib.rs

```diff
@@ -329,3 +332,81 @@ pub fn download_snapshot_archive<'a, 'b>(
         desired_snapshot_hash.0, rpc_addr
     ))
 }
+
+/// Checks if the given RPC node has a newer incremental snapshot available
+/// Returns Some(slot, hash) if a newer slot is available
+/// Returns None otherwise
+pub fn check_for_newer_incremental_snapshot(
+    rpc_addr: SocketAddr,
+    full_snapshot_slot: u64,
+    incremental_snapshot_slot: u64,
+) -> Option<(u64, Hash)> {
+    let incremental_redirect_url =
+        &format!("http://{}{}", rpc_addr, INCREMENTAL_SNAPSHOT_REQUEST_PATH);
+    let response = reqwest::blocking::get(incremental_redirect_url);
+    let response = match response {
+        Ok(r) => r,
+        Err(_e) => {
+            return None;
+        }
+    };
```

Comment:
Sorry for the flyby, but just wanted to let you know you can make these matches "rustier" and simpler to read by using [ok()](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok)

```suggestion
    let response = reqwest::blocking::get(incremental_redirect_url).ok()?;
```

Reply:
Thanks for the rustier feedback. I was tinkering with this trying to find less verbose solutions.

In this case I'm not sure if ok() will work since I'm trying to do an early return from the function rather than just setting response=None. I'll play around with it and see if there's a cleaner solution using ok()

Reply:
Note the question mark operator I also added. That will do the early return on None. Trying to find a good doc to link you to...

Reply:
https://doc.rust-lang.org/reference/expressions/operator-expr.html#the-question-mark-operator

Reply:
Ah yep I understand now. Thank you!